Makefile

```diff
@@ -1,43 +1,151 @@
--include Makefile.devhelpers
+SHELL := /usr/bin/env bash
 
-clean:
-	rm -rf venv
-	python3 -m venv venv
+# Want to skip this check? make CONTAINER_ENGINE=skip <target>
+# It can be anything, but bear in mind that commands want to be run with that first
+CONTAINER_ENGINE ?= $(shell which podman docker | head -n 1) 
+ifndef CONTAINER_ENGINE
+$(error CONTAINER_ENGINE is unset because it couldn't be auto-set; podman or docker missing from $$PATH. Skip this check with make CONTAINER_ENGINE=skip; but container commands will not work)
+endif
 
+# Don't want uv to use its local cache?
+# make IGNORE_UV_CACHE_FLAG=-n <target>
+IGNORE_UV_CACHE_FLAG ?= 
 
-install: clean
-	. venv/bin/activate && \
-	python setup.py install
+include Makefile.devhelpers
 
+# Attempt to auto set LDFLAGS
+# MacOS needs to use homebrew to install openssl and then use these LDFLAGS.
+# But, Linux typically doesn't need to change them, so don't. No idea what to do
+# for Windows
+UNAME_S := $(shell uname -s)
+ifeq ($(UNAME_S),Darwin)
+	LDFLAGS ?= "-I/opt/homebrew/opt/openssl/include -L/opt/homebrew/opt/openssl/lib"
+endif
```

Comment:
please note this is not needed after upgrade to python 3.11+

see https://github.com/app-sre/qontract-reconcile/pull/3826 and https://github.com/app-sre/qontract-reconcile/pull/4790/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5